app/policies/api/rubygem_policy.rb

```diff
@@ -9,22 +9,32 @@ def index?
   end
 
   def create?
-    api_key_scope?(:push_rubygem, rubygem)
+    mfa_requirement_satisfied? &&
```

Comment:
why not pass the `rubygem` here?

Reply:
There is no rubygem yet when this is called. It is found in the Pusher. I was planning to make a different push? action that handles that part separately.

Reply:
but you refer to the rubygem on the next line?

Reply:
Oops, you're right. I had written this for places where the class is passed:

```ruby
    rubygem = nil unless rubygem.is_a?(Rubygem)
```

I'll remove that call as well. Thank you for catching this.

Reply:
Updated to remove rubygem in both places and remove that check. I think I added that guard because I knew that _could_ happen and it fixed the bug, but I should have traced it back to the actual cause, which you found.